### PR TITLE
Remove requirement to sign-in with FxA

### LINKED ISSUE #1047: Remove sign-in requirement in frontend app
- [ ] [The `me` API resource on the server](https://github.com/mozilla/testpilot/blob/master/testpilot/users/views.py#L20) lists installed experiments and profile data for the logged in user - we'll no longer have this data or a login
- [ ] [The `me` model on the frontend](https://github.com/mozilla/testpilot/blob/master/testpilot/frontend/static-src/app/models/me.js) fetches some of its data from the auth'd user's profile on the server. Remove these parts and rely only on what the add-on knows - i.e. experiment installation state, client UUID, etc.

We should also treat presence of the add-on now as basically the "logged in" state.

### PR DESCRIPTION
**Pull request by lmorchard**
This is a big one. I did it in a few steps, so I left those as separate commits for now. But, here's what's happening:
- Change landing page call-to-action to add-on install button
- No more sign-in button on landing page
- Only require presence of add-on to see /experiments
- Add-on presence now enables "get started" / "manage" buttons in
  experiment listings
- Add-on presence indicates active user for header settings button
- Don't attempt to fetch server /api/me on frontend app startup
- Increase time to wait for response from add-on to messages
- Add more details about timeout failure while waiting for message
  responses from add-on
- Hardcode add-on installation URL to `/static/addon/addon.xpi` because
  /api/me is no longer available to configure that
- Remove django-allauth URL routes for sign in / out
- Remove the /api/me viewset
- Clean up related tests

Fixes #1047

### CODE REVIEW

**Comment by lmorchard** [comment]
Have to run for the day because I'm starving & eyes have glazed over. But I _think_ this might be in good shape? Needs a serious beating by someone other than me to find what I might have missed. I'll probably scour through it again tomorrow to see if I missed anything after I've slept on it.

**Comment by lmorchard** [comment]
Also note that the landing page messaging is still a little wonky with this PR. We've got issue #1041 to clean up the calls-to-action and whatnot.

File: testpilot/frontend/static-src/app/main.js [hunk 1]
```diff
@@ -65,9 +65,11 @@ app.extend({
   },
 
   // Send webChannel message to addon, use a Promise to wait for the answer.
-  waitForMessage(type, data, timeout = 1000) {
+  waitForMessage(type, data, timeout = 5000) {
```
Comment (ghost) [inline testpilot/frontend/static-src/app/main.js:68]: why the increase?

Comment (lmorchard) [inline testpilot/frontend/static-src/app/main.js:68]: I suspect, but can't be sure, that this was contributing to some of the "never finishes installing" errors on some machines. If the add-on took more than a second to finish installing & set up, the "Installing" barber poll button would never stop. Just thought of that while I was in here poking at things, so figured I'd give it a bump. An alternative might be to not have a timeout here at all and just have it continue to wait and never error with a timeout.

**Comment by lmorchard** [comment]
Added another commit to also cover issue #1041 in this PR

File: testpilot/frontend/static-src/app/views/experiment-row-view.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ export default BaseView.extend({
       hook: 'show-detail',
       name: 'active'
     },
-    'loggedIn': {
+    'hasAddon': {
       type: 'booleanClass',
       hook: 'show-detail',
       name: 'logged-in'
```
Comment (meandavejustice) [inline testpilot/frontend/static-src/app/views/experiment-row-view.js]: should this prop be updated to `has-addon`?

File: testpilot/frontend/static-src/test/views/landing-page.js [hunk 3]
```diff
@@ -57,7 +57,7 @@ test('Expected sendToGA call made when logged out', t => {
 
 test('Expected sendToGA call made with no experiments', t => {
   t.plan(3);
-  app.me.user.id = 'gary@busey.net';
+  app.me.hasAddon = true;
   app.sendToGA = (name, data) => {
     t.ok(data.dimension1 === true, 'is logged in');
```
Comment (meandavejustice) [inline testpilot/frontend/static-src/test/views/landing-page.js]: message should read 'addon is installed'

File: testpilot/frontend/static-src/test/views/landing-page.js [hunk 4]
```diff
@@ -30,21 +30,21 @@ test('Landing Page view renders', t => {
...
 
-  t.ok(view.loggedIn);
+  t.ok(view.hasAddon);
 });
 
-test('Expected sendToGA call made when logged out', t => {
+test('Expected sendToGA call made with no add-on', t => {
   t.plan(3);
   app.sendToGA = (name, data) => {
     t.ok(data.dimension1 === false, 'is not logged in');
```
Comment (meandavejustice) [inline testpilot/frontend/static-src/test/views/landing-page.js]: message needs updated

File: testpilot/frontend/static-src/test/views/landing-page.js [hunk 5]
```diff
@@ -69,11 +69,11 @@ test('Expected sendToGA call made with no experiments', t => {
...
-    {'addon_id': '@experiment1'},
-    {'addon_id': '@experiment2'}
-  ];
+  app.me.hasAddon = true;
+  app.me.installed = {
+    '@experiment1': {},
+    '@experiment2': {}
+  };
   app.sendToGA = (name, data) => {
     t.ok(data.dimension1 === true, 'is logged in');
```
Comment (meandavejustice) [inline testpilot/frontend/static-src/test/views/landing-page.js]: message needs updated

File: testpilot/frontend/static-src/test/views/landing-page.js [hunk 6]
```diff
@@ -57,7 +57,7 @@ test('Expected sendToGA call made when logged out', t => {
 
 test('Expected sendToGA call made with no experiments', t => {
   t.plan(3);
-  app.me.user.id = 'gary@busey.net';
```
Comment (meandavejustice) [inline testpilot/frontend/static-src/test/views/landing-page.js:60]: always sad to lose a reference to gary busey

File: testpilot/settings.py [hunk 7]
```diff
@@ -509,7 +509,8 @@ def lazy_langs():
     "https://code.cdn.mozilla.net/fonts/fira.css":
         "sha384-APhs/OUouhH+ZbBANL3+7a5J1sVSYyfUhGxTWiDMkTuEIO5fXWZonzNEj+CagDB7",
 }
-STATICINTEGRITY_CACHE_TTL = 30
+STATICINTEGRITY_CACHE_TTL = config('STATICINTEGRITY_CACHE_TTL',
```
Comment (meandavejustice) [inline testpilot/settings.py:512]: could you explain this change somewhere? is this just removing the hardcoded value?

File: testpilot/urls.py [hunk 8]
```diff
@@ -4,21 +4,16 @@
...
-from .users import views as users_views
 
 # Allow apps to contribute API parts
 router = routers.DefaultRouter(trailing_slash=False)
 experiment_views.register_views(router)
-users_views.register_views(router)
 
 urlpatterns = patterns(
     '',
     # users app is special, because it handles /accounts and /users
```
Comment (meandavejustice) [inline testpilot/urls.py]: comment needs to be updated now that we've removed `/users`

**Comment by meandavejustice** [comment]
I see that we aren't just removing the User model entirely, why is this?

**Comment by meandavejustice** [comment]
Although it's a little nitpicky, I think we should update the `.logged-in` classname in `testpilot/frontend/static-src/styles/modules/_experiment-list.scss` and corresponding templates in this pr, otherwise it may never be addressed again.

**Comment by meandavejustice** [comment]
We still have copy about being logged-in to fxa in `docs/FAQs.md` and `docs/DEPLOYMENT-VERIFICATION.md`

**Comment by meandavejustice** [comment]
This works and looks great! thanks for doing such a thorough job on this @lmorchard 
I'm happy to merge after the above comments are addressed

**Comment by lmorchard** [comment]
> I see that we aren't just removing the User model entirely, why is this?

We're still using the UserProfile model for the contributors list (#1046 / #1034)

File: testpilot/settings.py (hunk 7 above)
Comment (lmorchard) [inline testpilot/settings.py:512]: Yeah, just making the value configurable, so I can set it to a very short duration on dev where the static files change very often during development

**Comment by lmorchard** [comment]
I think I caught all the comments in this new commit, and squashed some things together

**Comment by meandavejustice** [comment]
Looks great to me, once tests pass I'll go ahead and merge

**Pull request merged**